homeassistant/components/fritzbox/coordinator.py

```diff
@@ -80,12 +87,16 @@ def _update_fritz_devices(self) -> FritzboxCoordinatorData:
                 device.present = False
 
             device_data[device.ain] = device
+            if device.ain not in self.data.devices:
+                self.new_devices.add(device.ain)
 
         template_data = {}
         if self.has_templates:
             templates = self.fritz.get_templates()
             for template in templates:
                 template_data[template.ain] = template
+                if template.ain not in self.data.templates:
+                    self.new_templates.add(template.ain)
 
```

Comment:
Just calculate the new data here instead.

self.new_templates = template_data.keys() -self.data.templates.keys()
self.new_devices = devices_data.keys() - self.data.devices.keys()